olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -363,13 +371,58 @@ bool DefaultCacheImpl::PromoteKeyLru(const std::string& key) {
   return true;
 }
 
+bool DefaultCacheImpl::PutMutableCache(const std::string& key,
+                                       const leveldb::Slice& value,
+                                       time_t expiry) {
+  if (!mutable_cache_) {
+    return true;
+  }
+
+  // can't put new item if cache is full and eviction disabled
+  const auto item_size = value.size();
+  const auto expected_size = mutable_cache_data_size_ + item_size + key.size() +
+                             key.size() + kExpirySuffixLength +
+                             kExpiryValueSize;
+  if (!mutable_cache_lru_ && expected_size > settings_.max_disk_storage) {
+    return false;
+  }
+
+  uint64_t data_size_change = 0u;
+  if (IsExpiryValid(expiry)) {
+    auto expiry_data_size = StoreExpiry(key, *mutable_cache_, expiry);
+    if (expiry_data_size == 0) {
+      return false;
+    }
+    data_size_change += expiry_data_size;
+  }
```

Comment:
For next PR!
I would propose to save expiry only after you store the real data, and maybe do it in one batch to save writes.